tests/pallas/pallas_test.py

```diff
@@ -744,6 +746,17 @@ def test_pallas_call_index_map_wrong_number_of_results(self):
         "Index map for input\\[0\\] must return 1 values to match .*Currently returning 2 values."):
       f(a)
 
+  def test_pallas_call_index_map_captures_consts(self):
```

Comment:
Can you also add a test with vmap, please? 

I think the args processing logic in batching rule needs updating as well.

Reply:
You were right, the batching rule needed changes. I added a test in pallas_vmap_test.py and I expanded the tests in tpu_pallas_test.py.